src/client/driver/utils/element-utils.js

```diff
@@ -12,7 +12,7 @@ export function exists (el) {
 }
 
 export function visible (el) {
-    if (!domUtils.isDomElement(el) && !domUtils.isTextNode(el))
+    if (!exists(el) || !domUtils.isDomElement(el) && !domUtils.isTextNode(el))
```

Comment:
Unfortunately, it will not work. `Selector.visible` needs an element because it is defined as a DOM node snapshot property:

https://github.com/DevExpress/testcafe/blob/cfa8d81a3ef8c4a8e33d6887394c3decb00115e0/src/client-functions/selectors/snapshot-properties.js#L17

https://github.com/DevExpress/testcafe/blob/cfa8d81a3ef8c4a8e33d6887394c3decb00115e0/src/client/driver/command-executors/client-functions/selector-executor/node-snapshots.js#L60

So, if there is no element, there will be no snapshot and `Selector.visible` will fail with an error.

We need to transform `Selector.visible` in a property getter like `Selector.exists`:

https://github.com/DevExpress/testcafe/blob/cfa8d81a3ef8c4a8e33d6887394c3decb00115e0/src/client-functions/selectors/add-api.js#L256

Reply:
OK I'll look into that tonight thanks